### DISCUSSION TITLE
It is possible to use Liquidjs generate variable and use it in Shopify Liquid?

### DISCUSSION THREAD

**Discussion by johnliu-code** [harttle/liquidjs#319] [Working with Shopify]
Hello,
I am new to liquid and seeing the way to check screen size and make a condition for upload different images.
But we could not call any JS variable to Liquid.
When I did research and find Liquidjs.
So, I am wondering if we could use Liquidjs to create a variable and use it in Shopify Liquid conditions?
I have tried to figure it out, but did not find the way how they work together.
Could you please help me out or give me some guide?
Big thanks!

**Comment by johnliu-code** [comment]
Here is something I have tried in Liquidjs Demo:

It works, but how can I bring this code into Shopify? Or if it is possible to do this in Shopify? Thanks!

const template = document.querySelector('[type="text/template"]')
const result = document.querySelector('#result')
const engine = new liquidjs.Liquid()

const screenAvailWidth = window.screen.availWidth;

engine
	.parseAndRender(template.innerHTML, {name: screenAvailWidth })
	.then(html => result.innerHTML = html)

<script type="text/template">
{% assign screen_size = name %}

{% assign is_mobile = false %}
{% if screen_size <= 600 %}
  This is a  mobile.
 {% assign is_mobile = true %}
 
{% elsif screen_size > 600 %}
{% assign is_mobile = false %}
  This is not a  mobile.
  <p> Screen Width = {{ screen_size }} px. </p>
{% else %}
{% endif %}
</script>

<div id="result"></div>

**Reply by johnliu-code** [reply]
https://jsfiddle.net/86gyjos5/

code link

**Reply by harttle** [reply]
LiquidJS is intended to be an Shopify liquid equivalent in JavaScript world.

That means it's used to port Shopify templates to JavaScript servers or apps easily. But LiquidJS and Shopify are not interoperable.
